Merging development branches

todd-a-jacobs opened the pull request:
I'm trying to forward-port various patches to your master branch, and have built a special tracking branch for your code. Please review this and future patches, and let's see how much of our trees we can merge together.

bmfs committed:
Added ti sync

bmfs committed:
minor change

bmfs committed:
changed ti sync

bmfs committed:
changed ti sync

bmfs committed:
changed ti sync

bmfs committed:
* Implemented 'ti sync'

The ticgit repository needs to be remote and tracked.
In the original repository do something like:
1) ti  # to create the ticgit branch
2) git push origin ticgit # to push it to remote
3) git branch -D ticgit
4) git checkout --track -b ticgit origin/ticgit
5) git checkout master # in my case is branch I usually work on
6) use ticgit normally and call 'ti sync' when you want to sync with the remote repository

When you colaborate you do something like:
1) git clone git://(...)  # clone the repository you want
2) git checkout --trac -b ticgit origin/ticgit
3) git checkout master # in my case is branch I usually work on
4) use ticgit normally and call 'ti sync' when you want to sync with the remote
repository

Commit:
Adds gem_installer task as dependency of setup task.

This commit addresses the situation where 'rake setup' fails due to
GemInstaller being undefined.

jeffWelling committed:
Updated project info to use my info, added myself to authors file

jeffWelling committed:
Deleted CHANGELOG - Why put effort into keeping a changelog when you could just as easily check the git revision history?

jeffWelling committed:
Updated gemspec

jeffWelling committed:
Added prerequisite section from Todd A. Jacobs' README - would have just applied a patch but it does not apply cleanly

jeffWelling committed:
Merge branch 'master' of http://github.com/pheres/ticgit into deving_merge_pheres

jeffWelling committed:
Merge branch 'master' of http://github.com/cthachuk/ticgit into deving_merge_pheres

jeffWelling committed:
Merge branch 'deving_merge_pheres' into deving

jeffWelling committed:
Merged master into deving

todd-a-jacobs committed:
Added support for non-alphanumeric titles.

- Incorporated ticket_title patch from Debian maintainer Michael
  Schutte. [git://git.debian.org/~michi-guest/ticgit.git @ bcaff2c]
- Closes Debian bug 514282.
  [http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=514282]

todd-a-jacobs committed:
Automatically choose system/source library.

- bin/ti learned to detect if it was running inside a Git source tree.
- bin/ti now automatically prefers the source version of the ticgit
  library when run from inside the Git tree.
- Made path names OS-agnostic within by using File.join instead of
  hard-coded directory separators.

todd-a-jacobs committed:
Prepend source tree lib directory to LOAD_PATH.

Add the ticgit.rb library from the source tree to the front of the load
path. This allows ticgitweb to run without first creating and installing
a ticgit gem, which is important when testing multiple branches of
development.

todd-a-jacobs committed:
Munge LOAD_PATH to prefer source tree for ticgit.

- bin/ti learned an easier way to prefer the ticgit.rb library in the
  source tree over an installed gem.
- bin/ti now uses same trick as bin/ticgitweb (see commit 253f646).

todd-a-jacobs committed:
Simplification of gemspec.

- Removed optional fields, especially those prone to clobbering between
  forks. This should make maintaining the gemspec among multiple clones
  easier.
- Added dynamic date. Date will be updated on each rebuild of the gem.
- Added dynamic versioning based on seconds since epoch. Version will be
  incremented on each rebuild of the gem.

jeffWelling commented:
Sounds great!
My current problem is that Im suffering from a laptop failure and have to replace it before I am able to make changes.  This should be done fairly quickly though so look for this to be merged as early as this coming week.

Commit:
Merge branch 'merging_codegnome' into deving

todd-a-jacobs committed:
Rakefile rewritten: bundler & rspec-2 added.

- Rakefile was rewritten to use bundler to handle dependencies.
- Gemfile and Gemfile.lock added to support bundler.
- Rake now builds the ticgit gem with a simplified ticgit.gemspec that
  will reduce merge conflicts.
- Test task rewritten to use rspec 2.0 instead of 1.3.0.
- The RSpec task was simplified, but currently fails to run the tests
  written for bacon.
- Default task is now to build and install a ticgit gem.

todd-a-jacobs committed:
Rewound bacon spec files to RSpec 1.3.0 tests.

- Replaced spec/ directory with the contents from
  e2a0853642175388ab282d8b64a14ef24c6b8d29 to remove bacon dependencies.
- Rake now runs test:spec using RSpec-2.
- 27 tests in suite: 3 pending and 3 failing.

todd-a-jacobs committed:
Removed AUTHORS file.

- According to the GNU guidelines, AUTHORS is intended for situations
  where "[t]he following contributions warranted legal paper exchanges
  with the Free Software Foundation."
- Counting contributions by commits is just an exercise in vanity, and
  doesn't merit special consideration.
- A MAINTAINERS, CONTRIBUTORS, or THANKS file could be added in the
  future, but should be left until the codebase stabilizes.

todd-a-jacobs committed:
spec_helper learned about mktmpdir

- Fixed potential race condition in TicGitSpecHelper::setup_new_git_repo
  between the time a tempfile was created and a same-named directory was
  created by using the mktmpdir method instead.
- Improved clarity of variable name for temporary directory.

jeffWelling commented:
Already merged :)
